paywall/src/components/checkout/Checkout.tsx

```diff
@@ -0,0 +1,109 @@
+import styled from 'styled-components'
+import React from 'react'
+import { RoundedLogo } from '../interface/Logo'
+
+import { UILocks, PaywallConfig, Account } from '../../unlockTypes' // eslint-disable-line no-unused-vars
+import CheckoutLock from './CheckoutLock'
+
+interface Props {
+  locks: UILocks
+  config: PaywallConfig
+  account: Account
+  purchase: (...args: any[]) => any
+}
+
+export const Checkout = ({
+  locks,
+  config,
+  account,
+  // keys,
+  purchase,
+}: Props) => {
+  const hasValidKey = Object.keys(locks).reduce((hasValidKey, address) => {
```

Comment:
this can become:

```js
const hasValidKey = Object.values(locks).filter(lock => lock.key.status === 'valid').length
```

Reply:
https://stackoverflow.com/questions/42966362/how-to-use-object-values-with-typescript/42967397#comment90677819_42967397 may help us here

Reply:
In the end, is that cleaner than using `Object.keys`?

Reply:
no. let me rephrase using Object.keys:

```js
const hasValidKey = Object.keys(locks).reduce((isValid, address) => locks[address].key.status === 'valid', false)
```

Don't do any calculation

Reply:
Oh... that's what you meant, using the key status rather than the expiration! Sorry! Yeah, I'll do that.